src/fetch.ts

```diff
@@ -105,16 +105,23 @@ export function createFetch (globalOptions: CreateFetchOptions): $Fetch {
         ctx.request = withQuery(ctx.request, ctx.options.params)
       }
       if (ctx.options.body && isPayloadMethod(ctx.options.method)) {
+        ctx.options.headers = new Headers(ctx.options.headers)
+
+        // Default to application/json for content-type and accept,
+        // when not explicitly set, and the body is JSON serializable.
         if (isJSONSerializable(ctx.options.body)) {
-          ctx.options.body = JSON.stringify(ctx.options.body)
```

Comment:
Can't we keep check inside this branch?

Reply:
Separating the logic to set default headers, from the logic to automatically stringify JSON bodies, makes the code more readable in my opinion.

Reply:
We could directly stringify here but make an exception for type: string. And remove header checking logic.

Reply:
Is there any reason to stringify request bodies that have a different content type than application/json?

From my experience, implicitly stringifying leads to more issues than it solves.

Reply:
Probably not. But then what would `options.body` value that is both serializable and we don't intend to serialize?

Reply:
For example when you want to send raw strings or numbers that need to be send unmodified to the server.

Reply:
Yes strings should be fixed as is. That's for the linked issue. JSON.stringify on other primitives does nothing. `JSON.stringify(2)` is `2` for instance.

Reply:
So, instead of my proposed changes, you want to adjust the `isJSONSerializable` function to return `false` for strings? Is that correct?

Reply:
I would still add application/json as default content type for string body if not specified. We could add a simple inline check like `typeof s === 'string' ? s : JSON.stringify(s)`